### PR TITLE
Add CUE file generator

### PR DESCRIPTION
**Pull request by DomBlack**
This commit adds the ability for Encore to generate CUE files based on calls to `config.Load`
within their services. The generated CUE files can be used with the standard CUE tooling to
create and validate the config needed for the app.

We support both `cue` and `json` tags being used on config structs.

### CODE REVIEW

File: compiler/internal/cuegen/testdata/basic_config_svc.cue [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+// are changed.
+//
+// For more information about this file, see:
+// https://encore.dev/docs/develop/config
+package svc
+
+Name:     string // The users name
+Port:     uint16
+ReadOnly: bool // true if we're in read only mode
+// MagicNumber is complicated and requires
```
Comment (ekerfelt) [inline compiler/internal/cuegen/testdata/basic_config_svc.cue:15]: doesn't seem so complicated to me :)

**Review by ekerfelt — APPROVED** [review]

File: compiler/internal/cuegen/service.go [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+		return err
+	}
+
+	fields, err := s.structToFields(concrete)
+	if err != nil {
+		return err
+	}
+
+	for _, fieldDecl := range fields {
+		field := fieldDecl.(*ast.Field)
```
[the remark this answers is not in the document]
Comment (DomBlack) [reply compiler/internal/cuegen/service.go:58]: Left over from some old code, changed the return type of structToFields so we don't need the cast

File: compiler/internal/cuegen/service.go [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+		// the same service to either the same or different struct types
+		// with both have the same field name
+		if existing, found := s.fieldLookup[name]; found {
+			if len(field.Comments()) > 0 {
+				if len(existing.Comments()) == 0 {
+					existing.SetComments(field.Comments())
+				} else {
+					existingCommentGrp := existing.Comments()[0]
+					existingCommentGrp.Position = 0 // now multi line, so needs to go above the field
+					for _, comment := range field.Comments() {
```
[the remark this answers is not in the document]
Comment (DomBlack) [reply compiler/internal/cuegen/service.go:75]: Good catch, added some code to deduplicate comments

File: compiler/internal/cuegen/service.go [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+		}
+
+		field := &ast.Field{
+			Label: defIdent,
+			Value: fieldType,
+		}
+		if decl.Doc != "" {
+			addCommentToField(field, decl.Doc)
+		} else {
+			// If there isn't a doc, we want to force a new section
```
[the remark this answers is not in the document]
Comment (DomBlack) [reply compiler/internal/cuegen/service.go:166]: For prettyness sake of the generated code, without this you go from:

```cue
#ExtraConfig: {
	Foo: string
	Baz: bytes
}

#ExtraConfig_1: {
	Foo: string
	Bar: int
}
```

to 

```cue
#ExtraConfig: {
	Foo: string
	Baz: bytes
}
#ExtraConfig_1: {
	Foo: string
	Bar: int
}
```

[the remark this answers is not in the document]
Comment (ekerfelt) [reply compiler/internal/cuegen/service.go:173]: ahh, pretty!

**Review by ekerfelt — APPROVED** [review]

**Pull request merged**